fix: set proper tags for atlas dashboards

QuentinBisson opened the pull request:
Towards https://github.com/giantswarm/roadmap/issues/3318, this PR sets proper tags on our dashboards to make it clear that we own those dashboards.

Once this is merged then I will add the missing datasources to our dashboards


<!--
Changelog must always be updated.
-->

### Checklist

- [x] Update changelog in CHANGELOG.md in an end-user friendly language.

hervenicol commented:
That looks neat! However, I did not know of this tags convention.
Do we have a doc that explains it?

Also, concerning mixins-generated dashboards, we should make sure we set the proper labels next time we update them. Whether it's by mixins config, yq scripts or documentation.

QuentinBisson commented:
So we technically do not have any tag conventions. We added the owner tag at some point but that's the only one. Considering mimir (resp. loki) add the `mimir` (resp. `loki`) tags, I switched them to component:component_name but those are definitely not set in stone :) 

If @giantswarm/team-atlas is fine with them then I can add a documentation entry.

Regarding the mixins, the loki and mimir scripts have those tags already configured :)

Rotfuks commented:
I like it. lgtm

hervenicol commented on README.md:
```diff
@@ -2,21 +2,28 @@
-The dashboards located under `helm/dashboards` are the dashboards hosted on each management cluster's grafana.
-The "public" ones are accessible by the customer, and the "private" ones are only accessible by Giant Swarm employees.
-Those dashboards are currently in JSON and will eventually be replaced to jsonnet format.
+All dashboards should strive to have proper tags to facilitate their discovery.
```
We could simplify wording here
```suggestion
All dashboards should have proper tags to facilitate their discovery.
```

hervenicol reviewed (approved):
Apart from one small remark I like it! :heart:
